Patterns/Chinese/Chinese-DateTime.yaml

```diff
@@ -7,9 +7,9 @@ MonthRegex: !simpleRegex
 DayRegex: !simpleRegex
   def: (?<day>01|02|03|04|05|06|07|08|09|10|11|12|13|14|15|16|17|18|19|20|21|22|23|24|25|26|27|28|29|30|31|1|2|3|4|5|6|7|8|9)
 DateDayRegexInChinese: !simpleRegex
-  def: (?<day>初一|三十|一日|十一日|二十一日|三十一日|二日|三日|四日|五日|六日|七日|八日|九日|十二日|十三日|十四日|十五日|十六日|十七日|十八日|十九日|二十二日|二十三日|二十四日|二十五日|二十六日|二十七日|二十八日|二十九日|一日|十一日|十日|二十一日|二十日|三十一日|三十日|二日|三日|四日|五日|六日|七日|八日|九日|十二日|十三日|十四日|十五日|十六日|十七日|十八日|十九日|二十二日|二十三日|二十四日|二十五日|二十六日|二十七日|二十八日|二十九日|十日|二十日|三十日|10日|11日|12日|13日|14日|15日|16日|17日|18日|19日|1日|20日|21日|22日|23日|24日|25日|26日|27日|28日|29日|2日|30日|31日|3日|4日|5日|6日|7日|8日|9日|一号|十一号|二十一号|三十一号|二号|三号|四号|五号|六号|七号|八号|九号|十二号|十三号|十四号|十五号|十六号|十七号|十八号|十九号|二十二号|二十三号|二十四号|二十五号|二十六号|二十七号|二十八号|二十九号|一号|十一号|十号|二十一号|二十号|三十一号|三十号|二号|三号|四号|五号|六号|七号|八号|九号|十二号|十三号|十四号|十五号|十六号|十七号|十八号|十九号|二十二号|二十三号|二十四号|二十五号|二十六号|二十七号|二十八号|二十九号|十号|二十号|三十号|10号|11号|12号|13号|14号|15号|16号|17号|18号|19号|1号|20号|21号|22号|23号|24号|25号|26号|27号|28号|29号|2号|30号|31号|3号|4号|5号|6号|7号|8号|9号)
+  def: (?<day>二十一日|三十一日|二十二日|二十三日|二十四日|二十五日|二十六日|二十七日|二十八日|二十九日|二十一日|三十一日|二十二日|二十三日|二十四日|二十五日|二十六日|二十七日|二十八日|二十九日|二十一号|三十 一号|二十二号|二十三号|二十四号|二十五号|二十六号|二十七号|二十八号|二十九号|二十一号|三十一号|二十二 号|二十三号|二十四号|二十五号|二十六号|二十七号|二十八号|二十九号|十一日|十二日|十三日|十四日|十五日| 十六日|十七日|十八日|十九日|十一日|二十日|三十日|十二日|十三日|十四日|十五日|十六日|十七日|十八日|十九日|二十日|三十日|10日|11日|12日|13日|14日|15日|16日|17日|18日|19日|20日|21日|22日|23日|24日|25日|26日|27日|28日|29日|30日|31日|十一号|十二号|十三号|十四号|十五号|十六号|十七号|十八号|十九号|十一号|二十号|三十号|十二号|十三号|十四号|十五号|十六号|十七号|十八号|十九号|二十号|三十号|10号|11号|12号|13号|14号|15号|16号|17号|18号|19号|20号|21号|22号|23号|24号|25号|26号|27号|28号|29号|30号|31号|初一|三十|一日|二日|三日|四日|五日|六日|七日|八日|九日|一日|十日|二日|三日|四日|五日|六日|七日|八日|九日|十日|1日|2日|3日|4日|5日|6日|7日|8日|9日|一号|二号|三号|四号|五号|六号|七号|八号|九号|一号|十号|二号|三号|四号|五号|六号|七号|八号|九号|十号|1号|2号|3号|4号|5号|6号|7号|8号|9号)
```

Comment:
Can you explain this change and the DayRegexNumInChinese one?

Reply:
> Can you explain this change and the DayRegexNumInChinese one?

This change fixed the issue #2381 . I changed their order, putted “三十” behind “三十一“ to avoid recognizing  ”五月三十一日“，”五月三十一“ as "五月三十”.

Reply:
Wouldn't it be better to simplify this a little? It seems to me that there's quite a bit of repetition.
For example for the case you said, why not use "五月三十(一?)" ? Similarly for quite a lot of cases.

Reply:
Yes. I think I can simplify it.